phantom-dsl/src/main/scala/com/outworkers/phantom/macros/RootMacro.scala

```diff
@@ -113,25 +118,36 @@ class RootMacro(val c: blackbox.Context) {
   case class MatchedField(
     left: Record.Field,
     right: Column.Field
-  ) extends RecordMatch
+  ) extends RecordMatch {
+    def column: Column.Field = right
+
+    def record: Record.Field = left
+  }
 
   implicit class ListMapOps[K, V, M[X] <: Traversable[X]](
     val lm: ListMap[K, M[V]]
   )(implicit cbf: CanBuildFrom[Nothing, V, M[V]]) {
 
     /**
-     * Every entry in this ordered map is a traversable of type [[M]].
-     * That means every key holds a sequence of elements.
-     * This function will remove the element [[elem]] from that sequence
-     * for the provided key.
-     */
-
+      * Every entry in this ordered map is a traversable of type [[M]].
+      * That means every key holds a sequence of elements.
+      * This function will remove the element [[elem]] from that sequence
+      * for the provided key.
+      */
     def remove(key: K, elem: V): ListMap[K, M[V]] = {
       lm.get(key) match {
```

Comment:
@iRhysBradburyOTRL Easier to read the way I did it, I was also considering that if the traversable for a key is left empty after a removal, than the key should be completely removed instead of pointing to an empty traversable.